packages/shared/src/hooks/useOpenShareTxDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import { makeStyles } from '@masknet/theme'
+import { resolveTransactionLinkOnExplorer, useChainId } from '@masknet/web3-shared-evm'
+import DoneIcon from '@mui/icons-material/Done'
+import { Link, Typography } from '@mui/material'
+import { FC, memo } from 'react'
+import { useShowConfirm } from '../contexts'
+import { useSharedI18N } from '../locales'
+
+const useStyles = makeStyles()((theme) => ({
+    content: {
+        textAlign: 'center',
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    icon: {
+        fontSize: 64,
+        width: 64,
+        height: 64,
+    },
+    link: {
+        marginTop: theme.spacing(0.5),
+    },
+    primary: {
+        fontSize: 18,
+        marginTop: theme.spacing(1),
+    },
+    secondary: {
+        fontSize: 14,
+    },
+}))
+
+interface ShareTransactionOptions {
+    title?: string
+    message?: string
+    content?: string
+    hash: string
+    buttonLabel?: string
+    // TODO Expose onShare until we have share API added our runtime
```

Comment:
Can you explain more clearly?

Reply:
We can't access `activatedSocialNetworkUI` outside `packages/mask` now,  although it should be a feature of a runtime for plugin.

I once introduced `runtime` at https://github.com/DimensionDev/Maskbook/pull/5632#issue-1128316587 .

```
<chain-runtime>
  <ui-runtime>
    <plugin-runtime>{children}</plugin-runtime>
  </ui-runtime>
</chain-runtime>
```

We can separate different concepts into individual layers, and we can implement each of them across platforms. iOS, android, and we might have desktop in the future, and even for mocking/testing.

If we implement mocking runtime, we can easily test with headless JS runtime. For example, we can turn UI interaction into function-call.

```
const share = () => Promise.resolve(true)
const pickToken = () => Promise.resolve({name: 'Mask'})
const web3Provider = () => connnectToLocalNetwork()
```

Reply:
You may be interested to add shareTo at `Plugin.SNSAdaptor.SNSAdaptorContext` at packages/plugin-infra/src/types.ts.

Reply:
ping

Reply:
I would like to do it within another patch.

Reply:
It's ok, please open an issue so we won't forget to do it in the future.

Reply:
> It's ok, please open an issue so we won't forget to do it in the future.

https://github.com/DimensionDev/Maskbook/issues/6233